pkg/framework/aries/api/vdri/vdri.go

```diff
@@ -113,6 +118,14 @@ func WithServiceEndpoint(serviceEndpoint string) DocOpts {
 	}
 }
 
+// WithRequestBuilder allows to supply request builder
+// which can be used to add headers to request stream to be sent to HTTP binding URL
+func WithRequestBuilder(builder func(payload []byte) (io.Reader, error)) DocOpts {
+	return func(opts *CreateDIDOpts) {
+		opts.RequestBuilder = builder
+	}
+}
```

Comment:
I'm wondering if this option is even needed at all. I would have assumed the specific VDRI implementation would take care of building the request and serializing whatever needs to be serialized. The VDRI implementation for the DID method can be picked with the `VDRI.Accept(method string)` method.

Reply:
we thought about the same when we started doing this change. SDK has feature of injecting custom VDRI implementation through opts/context. It cannot be done with controller APIs.

But this option can be used to modify request being sent to http binding URL.  REST API use this option to add http binding endpoint specific headers.

Reply:
@sudeshrshetty 

> It cannot be done with controller APIs.

Sounds like some kind of refactor is needed to support this use case?

Reply:
This is what this PR is about. HTTP Binding implementation creates DID with minimal features like public keys and service endpoint (from current agent) and submits it to http binding endpoint URL. This option will help us to add custom header, in long run we have to go as you suggested.

This PR is to unblock other public DID related features in controller API

Reply:
Will be handled as part of next PR for #860